Networking/NetworkingTests/Remote/CouponsRemoteTests.swift

```diff
@@ -350,6 +350,52 @@ final class CouponsRemoteTests: XCTestCase {
         let resultError = try XCTUnwrap(result.failure as? NetworkError)
         XCTAssertEqual(resultError, .unacceptableStatusCode(statusCode: 500))
     }
+
+    // MARK: - Retrieve coupon
+
+    /// Verifies that retrieveCoupon properly parses the `coupon` sample response.
+    ///
+    func test_retrieveCoupon_properly_returns_parsed_coupon() throws {
+        // Given
+        let sampleCouponID: Int64 = 720
+        let remote = CouponsRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "coupons/\(sampleCouponID)", filename: "coupon")
+
+        // When
+        let result = waitFor { promise in
+            remote.retrieveCoupon(for: self.sampleSiteID, couponID: sampleCouponID) { (result) in
+                promise(result)
+            }
+        }
+
+        // Then
+        XCTAssert(result.isSuccess)
+        let coupon = try XCTUnwrap(result.get())
+        let expectedCoupon = sampleCoupon()
+        XCTAssertEqual(coupon, expectedCoupon)
+    }
+
+    /// Verifies that retrieveCoupon properly relays Networking Layer errors.
+    ///
+    func test_retrieveCoupon_properly_relays_networking_errors() throws {
+        // Given
+        let sampleCouponID: Int64 = 720
+        let remote = CouponsRemote(network: network)
+        let error = NetworkError.unacceptableStatusCode(statusCode: 500)
+        network.simulateError(requestUrlSuffix: "coupons", error: error)
```

Comment:
For the [CI failure](https://app.circleci.com/pipelines/github/woocommerce/woocommerce-ios/17091/workflows/c023536d-d19f-4608-98a8-7a3effca435c/jobs/32925), I think it's from the request URL:
```suggestion
        network.simulateError(requestUrlSuffix: "coupons/\(sampleCouponID)", error: error)
```

Reply:
Updated in e040621416cacb14309167051a49382c1eef8207 👍